{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "f7aca370-7aa0-4cd0-8465-6e62cdd46d9d",
   "metadata": {},
   "source": [
    "# Seminar 11.11.2025"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fd6113a5-bfc7-415d-887b-93739245b522",
   "metadata": {},
   "source": [
    "## 1. Adding Jupyter Notebook to Github\n",
    "Add this Jupyter notebook to the GitHub repository you created in the last seminar!\n",
    "To do this, move the notebook to the corresponding folder. Use the ```add``` command to update the index using the current content found in the working tree, to prepare the content staged for the next commit. Then ```commit``` the content and create a meaningful message (```-m```). Finally, ```push```` your content to your remote repository. "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c958b537-23d3-4144-9fd1-8b0f02acf0d7",
   "metadata": {},
   "source": [
    "<div class=\"alert alert-block alert-warning\"> \n",
    "Remember that the password GitHub asks for is the token you created.\n",
    "</div>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1b0b6f7c-f453-481f-aafc-37afeb917926",
   "metadata": {},
   "source": [
    "## 2. Prepare todays data\n",
    "### 2.1 Create working directory\n",
    "Create a folder named “data” in your repository "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "9738b31a-bf41-4a7e-bea1-9bcd85d8538c",
   "metadata": {},
   "outputs": [],
   "source": [
    "mkdir data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "797078d3-a721-40b5-9278-1fef264cdee9",
   "metadata": {},
   "outputs": [],
   "source": [
    "cd ./data/"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3fb15964-9d44-431a-91b5-eb3961cb4553",
   "metadata": {},
   "source": [
    "### 2.2 Download data\n",
    "Download the two fastq-files for today’s lesson.  If necessary, you may need to move the data sets to the newly created folder."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "29d8c1f9-800f-44b4-b95e-65440dfd4f8f",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2025-11-11 10:23:10--  https://raw.githubusercontent.com/nischoen/BiodatenProzessierung/refs/heads/main/data/0_sup_notrim.fastq\n",
      "Auflösen des Hostnamens raw.githubusercontent.com (raw.githubusercontent.com)… 185.199.109.133, 185.199.110.133, 185.199.111.133, ...\n",
      "Verbindungsaufbau zu raw.githubusercontent.com (raw.githubusercontent.com)|185.199.109.133|:443 … verbunden.\n",
      "HTTP-Anforderung gesendet, auf Antwort wird gewartet … 200 OK\n",
      "Länge: 3674439 (3,5M) [text/plain]\n",
      "Wird in »0_sup_notrim.fastq« gespeichert.\n",
      "\n",
      "0_sup_notrim.fastq  100%[===================>]   3,50M  --.-KB/s    in 0,07s   \n",
      "\n",
      "2025-11-11 10:23:10 (51,0 MB/s) - »0_sup_notrim.fastq« gespeichert [3674439/3674439]\n",
      "\n"
     ]
    }
   ],
   "source": [
    "wget https://raw.githubusercontent.com/nischoen/BiodatenProzessierung/refs/heads/main/data/0_sup_notrim.fastq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "a3b38b41-6172-487e-aa2a-3db91479448b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2025-11-11 10:23:50--  https://raw.githubusercontent.com/nischoen/BiodatenProzessierung/refs/heads/main/data/0_sup_trimed.fastq\n",
      "Auflösen des Hostnamens raw.githubusercontent.com (raw.githubusercontent.com)… 185.199.109.133, 185.199.110.133, 185.199.111.133, ...\n",
      "Verbindungsaufbau zu raw.githubusercontent.com (raw.githubusercontent.com)|185.199.109.133|:443 … verbunden.\n",
      "HTTP-Anforderung gesendet, auf Antwort wird gewartet … 200 OK\n",
      "Länge: 3409003 (3,3M) [text/plain]\n",
      "Wird in »0_sup_trimed.fastq« gespeichert.\n",
      "\n",
      "0_sup_trimed.fastq  100%[===================>]   3,25M  --.-KB/s    in 0,07s   \n",
      "\n",
      "2025-11-11 10:23:51 (47,8 MB/s) - »0_sup_trimed.fastq« gespeichert [3409003/3409003]\n",
      "\n"
     ]
    }
   ],
   "source": [
    "wget https://raw.githubusercontent.com/nischoen/BiodatenProzessierung/refs/heads/main/data/0_sup_trimed.fastq"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7b2c037a-c79a-4db8-a0b1-4f2d5841d32a",
   "metadata": {},
   "source": [
    "### Renaming\n",
    "Rename the files respectively  to 'BC1_notrim.fastq' and BC1_trimed.fastq'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "37bb0968-0f21-4f68-9ec9-d512a7da49f7",
   "metadata": {},
   "outputs": [],
   "source": [
    "# your commands"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b85a3a4b-6e00-4ee5-91ed-e77ebf97edd8",
   "metadata": {},
   "source": [
    "## 3. Repeat: Quality control\n",
    "### 3.1 Overview\n",
    "Use wordcount (```wc```) to check how many reads are in each of the both fastq files!"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "ea2d08d0-410a-477d-8121-e011e799a84c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4004 ./BC1_notrim.fastq\n"
     ]
    }
   ],
   "source": [
    "wc -l ./BC1_notrim.fastq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "a190b9c1-cd34-4afb-b6b7-905393ae4be4",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "@02040615-3184-4a7b-9fb1-130e4bca8d82\tqs:f:16,7805\tch:i:45\tst:Z:2023-03-02T14:10:13.664000+00:00\tRG:Z:741dfaa4-41ba-469b-a04f-bc0c587b0122_dna_r10.4.1_e8.2_400bps_sup@v5.2.0_SQK-NBD114-24_barcode01\tPU:Z:AWJ804\tDT:Z:2023-03-02T14:01:04.294000+00:00\n",
      "GTTATGTTTGTAACATCTAAGATTACGTATTGTCAAGGTTAACACAAAGACACCGACAACTTTCTTCAGCACCTGTGATCCTGCAAGTAGTCATATGCTTGTCTCAAAGATTAAGCCATGCATGTCTTAGTGTAAATAACTCTTATGAAACTGCGAATGGCTCATTAAATCAGTTATCAAATACGTGATGGTCGCCTACTACATGGATAACCGTAGTAATTCTAGAGCTAATACATGCACAAAGGCCCACGCAAGTGGGTTGTGATTATTAGATCCGAACCAGCCTCGTTCTGGTGATTCATAATAATTGGTCGAACCGCATGCCCTCGTGGTGGCGGTGATTCATCCTTATTTAGGTGCTGAAGAAAGTTGTCGGTGTCTTTGTGTTAACCTTAGCAATACGTA\n",
      "+\n",
      "\"\"#$%&'''%%$$$$$#####$$%%'''())('&)))*./+++-0<=>ADCCCA???FGLFH?:8811KNJJJKKSNMMLEB22/-,,+***,,--.//;9966677;@EC::9;>GECDEENKLKJA@888JHHHIPPLKMLLKLPKGEFHB@--,,+,,,+1122458AFLOOSSSRRRRNONMJJJA>>965557>::322222<=KD@??:::55568<9:@ILSRRRJIGGGIIK:::9?AAAAAHIGEEGJPQNOLLIJHHFCC0,++**----./////EKKNPOKKKKLQOGNMKLKAAAAAGEEC>;;300111542.-00;CMNOODDD;;;9556,,,,,,.7,7889<<<<<HFIBA@?<8*)***1000-()))),,0AAA111124,,+*'\n",
      "@00558b02-857d-40b6-8dad-cb3ca2400435\tqs:f:16,7857\tch:i:122\tst:Z:2023-03-02T14:07:27.199000+00:00\tRG:Z:741dfaa4-41ba-469b-a04f-bc0c587b0122_dna_r10.4.1_e8.2_400bps_sup@v5.2.0_SQK-NBD114-24_barcode01\tPU:Z:AWJ804\tDT:Z:2023-03-02T14:01:04.294000+00:00\n",
      "GTTCCTCTACTCGTTCAGTTACGTATTGCTAAGGTTAACACAAAGACACCCGACAACTTTCTTCAGCACCTGGTGATCCTGCCAGTAGTCATATGCTTGTCTCAAAGACTAAGCCATGCATGTCTAAGTATAAGTATTATACAGCGAAACTGCGAATGGCTCATTAAAACAGTTATAGTTTACATGATAATCGAATTTACATGGATAACCGTGGTAATTCTAGAGCTAATCATGCTGGCGTGCCCGACTTTCAGAAGGCAGTATTTATTAGATAACAAGCCAATATTCCTTCGGGTCTATTGTGGTGATCCATAATAACTGATCGAATCGCAAGGCCCTTGGCCGCGATAAGTCATTCAAGTTTCTGCCCATCAGCTTTCGATGGTAGTGTATTGGACTACCATGGCGTTCACGGGTAACGGAGAATTAGGGTTCGATTCCGGAGAGGGAGCCTGAGAAACGGCTACCACATCTAAGGAAGGCAGCAGGCGCGTAAATTACCCAATCCCGACTCGGGGAGGTAGTATAGCAGATGACGGACCGGGGCCTTCGCAGGTACGGAATGCAATGAGCCGAATTTACACCCCTTAGCGAGGATCGATTGGAGGGCAAGTCTGGTGCCAGCAGCCGCGGTAATTCCAGCTCCAATAGCGTATATTAAAGTTGTTGCAGTTAAAAAGCTCGTAGTTGGGTTTCTGGGGGGAGTTCTTCGATCGCCATTGGCGATTAGAGGCTCCCTCCCATCCTTCTGTTAACGTTACTCGGCATTCATTTGCTGGTGGCGGGCTCAGATATTTTATCTTGAGAAAATTAGAGTGTTTCAGGCAGGCTAGGCCGGAATACATTAGCATGGAATAATGGAATAGGACTACGGTCTCTTTGTTGGTTTGAGGGACTGGAGTAATGATTAATAGGGATAGTTGGGGGCATTAGTATTTAATTGTCAGAGGTGGAATTCTCAGATTTGTTAAAGACTAACTTATGCGAAAGCATTTGCCAAGGATGTTTTCATTAATCAGGAACGAAAGTTAGGGGATCAAAGACGATCAGATACCGTCCTAGTCTTAACCATAAACTATGCCGACTCGGGATCGGGGGCGCGAACTATCCGCCTTCGGCACCGTATGAGAAATCAAAGTCTTTGGGTTCTGGGGGGAGTATGGTCGCAAGGCTGAAACTTAAAGGAATTGACGGAAGGGCACCACCAGGCGTGGAGCTTGCGGCTCAATTTGACTCAACACGGGAAAACTTACCAGGTCCAGACATAGTGAGGATTGACAGATTGATAGCTCTTTCTTGATTCTATGGGTGGTGGTGCATGGCCGTTCTTAGTTGGTGGAGTGATTTGTCCTGGTTAATTCCGTTAACGAACGAGACCTTTAGCCTGCTAACTAGTCTCAGCAATCTCAATTTGTGCAGGCTTCTAGAGGGACTTTGTGGCGCAAACACAAGGAAGTTTGAGGCAATAACAGGTCTGTGATGCCCTTAGATGTCCTGGGCCGCACGCGTGCTACACTGACACCTGCAGCGAGTACTTCCCAGCCCCGAGAGGCGGCTGGTAATCAGCAATAGGTGTCGTGATGGGGATAGATCTTTGGAATTATTGATCTTGAACGAGGAATGCCTAGTAAGCGCAGGTCATTCACTCATAGTCCTGCCCTTTGTACACACCGCCCGTCCGCTCCCTTACCGATTTCGAGTGGTCCGGTGAATCTTTCGGACTGCATATGGCCCCGTGCCATGTATGGAAAGTCAGGTAAACCTTATCACTGCGAGGAAGGAGAAGTCGTAACAAGGTTTCCGTAGGTGAACCTGCGGAAGGATCAGGTGCGACTATTTGAGGGGGTTGCTTTTGTGTTAACCTTAGCAATA\n"
     ]
    }
   ],
   "source": [
    "head -6 ./BC1_notrim.fastq #show only the 6 first rows of a file "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "97bf1290-3d2d-4fd1-a081-6812d64a501a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "@02040615-3184-4a7b-9fb1-130e4bca8d82\tqs:f:16,7805\tch:i:45\tst:Z:2023-03-02T14:10:13.664000+00:00\tRG:Z:741dfaa4-41ba-469b-a04f-bc0c587b0122_dna_r10.4.1_e8.2_400bps_sup@v5.2.0_SQK-NBD114-24_barcode01\tPU:Z:AWJ804\tDT:Z:2023-03-02T14:01:04.294000+00:00\n",
      "GTTATGTTTGTAACATCTAAGATTACGTATTGTCAAGGTTAACACAAAGACACCGACAACTTTCTTCAGCACCTGTGATCCTGCAAGTAGTCATATGCTTGTCTCAAAGATTAAGCCATGCATGTCTTAGTGTAAATAACTCTTATGAAACTGCGAATGGCTCATTAAATCAGTTATCAAATACGTGATGGTCGCCTACTACATGGATAACCGTAGTAATTCTAGAGCTAATACATGCACAAAGGCCCACGCAAGTGGGTTGTGATTATTAGATCCGAACCAGCCTCGTTCTGGTGATTCATAATAATTGGTCGAACCGCATGCCCTCGTGGTGGCGGTGATTCATCCTTATTTAGGTGCTGAAGAAAGTTGTCGGTGTCTTTGTGTTAACCTTAGCAATACGTA\n",
      "+\n",
      "\"\"#$%&'''%%$$$$$#####$$%%'''())('&)))*./+++-0<=>ADCCCA???FGLFH?:8811KNJJJKKSNMMLEB22/-,,+***,,--.//;9966677;@EC::9;>GECDEENKLKJA@888JHHHIPPLKMLLKLPKGEFHB@--,,+,,,+1122458AFLOOSSSRRRRNONMJJJA>>965557>::322222<=KD@??:::55568<9:@ILSRRRJIGGGIIK:::9?AAAAAHIGEEGJPQNOLLIJHHFCC0,++**----./////EKKNPOKKKKLQOGNMKLKAAAAAGEEC>;;300111542.-00;CMNOODDD;;;9556,,,,,,.7,7889<<<<<HFIBA@?<8*)***1000-()))),,0AAA111124,,+*'\n",
      "@00558b02-857d-40b6-8dad-cb3ca2400435\tqs:f:16,7857\tch:i:122\tst:Z:2023-03-02T14:07:27.199000+00:00\tRG:Z:741dfaa4-41ba-469b-a04f-bc0c587b0122_dna_r10.4.1_e8.2_400bps_sup@v5.2.0_SQK-NBD114-24_barcode01\tPU:Z:AWJ804\tDT:Z:2023-03-02T14:01:04.294000+00:00\n",
      "GTTCCTCTACTCGTTCAGTTACGTATTGCTAAGGTTAACACAAAGACACCCGACAACTTTCTTCAGCACCTGGTGATCCTGCCAGTAGTCATATGCTTGTCTCAAAGACTAAGCCATGCATGTCTAAGTATAAGTATTATACAGCGAAACTGCGAATGGCTCATTAAAACAGTTATAGTTTACATGATAATCGAATTTACATGGATAACCGTGGTAATTCTAGAGCTAATCATGCTGGCGTGCCCGACTTTCAGAAGGCAGTATTTATTAGATAACAAGCCAATATTCCTTCGGGTCTATTGTGGTGATCCATAATAACTGATCGAATCGCAAGGCCCTTGGCCGCGATAAGTCATTCAAGTTTCTGCCCATCAGCTTTCGATGGTAGTGTATTGGACTACCATGGCGTTCACGGGTAACGGAGAATTAGGGTTCGATTCCGGAGAGGGAGCCTGAGAAACGGCTACCACATCTAAGGAAGGCAGCAGGCGCGTAAATTACCCAATCCCGACTCGGGGAGGTAGTATAGCAGATGACGGACCGGGGCCTTCGCAGGTACGGAATGCAATGAGCCGAATTTACACCCCTTAGCGAGGATCGATTGGAGGGCAAGTCTGGTGCCAGCAGCCGCGGTAATTCCAGCTCCAATAGCGTATATTAAAGTTGTTGCAGTTAAAAAGCTCGTAGTTGGGTTTCTGGGGGGAGTTCTTCGATCGCCATTGGCGATTAGAGGCTCCCTCCCATCCTTCTGTTAACGTTACTCGGCATTCATTTGCTGGTGGCGGGCTCAGATATTTTATCTTGAGAAAATTAGAGTGTTTCAGGCAGGCTAGGCCGGAATACATTAGCATGGAATAATGGAATAGGACTACGGTCTCTTTGTTGGTTTGAGGGACTGGAGTAATGATTAATAGGGATAGTTGGGGGCATTAGTATTTAATTGTCAGAGGTGGAATTCTCAGATTTGTTAAAGACTAACTTATGCGAAAGCATTTGCCAAGGATGTTTTCATTAATCAGGAACGAAAGTTAGGGGATCAAAGACGATCAGATACCGTCCTAGTCTTAACCATAAACTATGCCGACTCGGGATCGGGGGCGCGAACTATCCGCCTTCGGCACCGTATGAGAAATCAAAGTCTTTGGGTTCTGGGGGGAGTATGGTCGCAAGGCTGAAACTTAAAGGAATTGACGGAAGGGCACCACCAGGCGTGGAGCTTGCGGCTCAATTTGACTCAACACGGGAAAACTTACCAGGTCCAGACATAGTGAGGATTGACAGATTGATAGCTCTTTCTTGATTCTATGGGTGGTGGTGCATGGCCGTTCTTAGTTGGTGGAGTGATTTGTCCTGGTTAATTCCGTTAACGAACGAGACCTTTAGCCTGCTAACTAGTCTCAGCAATCTCAATTTGTGCAGGCTTCTAGAGGGACTTTGTGGCGCAAACACAAGGAAGTTTGAGGCAATAACAGGTCTGTGATGCCCTTAGATGTCCTGGGCCGCACGCGTGCTACACTGACACCTGCAGCGAGTACTTCCCAGCCCCGAGAGGCGGCTGGTAATCAGCAATAGGTGTCGTGATGGGGATAGATCTTTGGAATTATTGATCTTGAACGAGGAATGCCTAGTAAGCGCAGGTCATTCACTCATAGTCCTGCCCTTTGTACACACCGCCCGTCCGCTCCCTTACCGATTTCGAGTGGTCCGGTGAATCTTTCGGACTGCATATGGCCCCGTGCCATGTATGGAAAGTCAGGTAAACCTTATCACTGCGAGGAAGGAGAAGTCGTAACAAGGTTTCCGTAGGTGAACCTGCGGAAGGATCAGGTGCGACTATTTGAGGGGGTTGCTTTTGTGTTAACCTTAGCAATA\n"
     ]
    }
   ],
   "source": [
    "# You can also use the Stream Editor (SED) to print certain lines to your command line.\n",
    "\n",
    "sed -n '1,6p' ./BC1_notrim.fastq \n",
    "\n",
    "# in this example we use -n to suppress automatic printing of all lines\n",
    "# then we select the row 1 to 6 \n",
    "# 'p' prints the selected row \n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8eefd0dd-672b-4940-990c-c73e46563cde",
   "metadata": {},
   "source": [
    "A Fastq file always consists of four lines. The first line (allways starts with '@')is the annotation line of the read, followed by the line containing the actual nucleotide sequence. The third line contains a ‘+’ and the fourth line contains the quality of each nucleotide from line two."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e828517e-a284-4f4d-9e14-a02191f974a8",
   "metadata": {},
   "source": [
    "The easiest way to get the correct number of reads would be to divide the output from ```wc``` by 4 but you can also use SED again to count only every 4th line"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "a6d2c35a-a458-43f2-9f5f-7e0b15c9fc28",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1001\n"
     ]
    }
   ],
   "source": [
    "sed -n '1~4p' ./BC1_notrim.fastq | wc -l\n",
    "\n",
    "# 1~4 selects only every 4th line starting from the first line"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "d6975037-ead5-4cc1-97dd-39f0c83249f8",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "@02040615-3184-4a7b-9fb1-130e4bca8d82\tqs:f:16,7805\tch:i:45\tst:Z:2023-03-02T14:10:13.664000+00:00\tRG:Z:741dfaa4-41ba-469b-a04f-bc0c587b0122_dna_r10.4.1_e8.2_400bps_sup@v5.2.0_SQK-NBD114-24_barcode01\tPU:Z:AWJ804\tDT:Z:2023-03-02T14:01:04.294000+00:00\n",
      "@00558b02-857d-40b6-8dad-cb3ca2400435\tqs:f:16,7857\tch:i:122\tst:Z:2023-03-02T14:07:27.199000+00:00\tRG:Z:741dfaa4-41ba-469b-a04f-bc0c587b0122_dna_r10.4.1_e8.2_400bps_sup@v5.2.0_SQK-NBD114-24_barcode01\tPU:Z:AWJ804\tDT:Z:2023-03-02T14:01:04.294000+00:00\n",
      "@03bdaa96-650e-429d-a0e1-bfea23cff789\tqs:f:22,1646\tch:i:107\tst:Z:2023-03-02T14:05:38.083000+00:00\tRG:Z:741dfaa4-41ba-469b-a04f-bc0c587b0122_dna_r10.4.1_e8.2_400bps_sup@v5.2.0_SQK-NBD114-24_barcode01\tPU:Z:AWJ804\tDT:Z:2023-03-02T14:01:04.294000+00:00\n",
      "@03cc2f3b-099b-436f-aeb8-d04a257b76da\tqs:f:16,3081\tch:i:27\tst:Z:2023-03-02T14:10:44.906000+00:00\tRG:Z:741dfaa4-41ba-469b-a04f-bc0c587b0122_dna_r10.4.1_e8.2_400bps_sup@v5.2.0_SQK-NBD114-24_barcode01\tPU:Z:AWJ804\tDT:Z:2023-03-02T14:01:04.294000+00:00\n",
      "@03f6c00e-c430-416c-bfdc-be0ed61ebe20\tqs:f:16,336\tch:i:114\tst:Z:2023-03-02T14:08:26.759000+00:00\tRG:Z:741dfaa4-41ba-469b-a04f-bc0c587b0122_dna_r10.4.1_e8.2_400bps_sup@v5.2.0_SQK-NBD114-24_barcode01\tPU:Z:AWJ804\tDT:Z:2023-03-02T14:01:04.294000+00:00\n",
      "@040c8038-7668-438f-8850-e070bdcf5527\tqs:f:16,6742\tch:i:83\tst:Z:2023-03-02T14:11:51.341000+00:00\tRG:Z:741dfaa4-41ba-469b-a04f-bc0c587b0122_dna_r10.4.1_e8.2_400bps_sup@v5.2.0_SQK-NBD114-24_barcode01\tPU:Z:AWJ804\tDT:Z:2023-03-02T14:01:04.294000+00:00\n",
      "@01692550-d71f-44b2-91d2-67d06725d027\tqs:f:21,3146\tch:i:107\tst:Z:2023-03-02T14:18:23.182000+00:00\tRG:Z:741dfaa4-41ba-469b-a04f-bc0c587b0122_dna_r10.4.1_e8.2_400bps_sup@v5.2.0_SQK-NBD114-24_barcode01\tPU:Z:AWJ804\tDT:Z:2023-03-02T14:01:04.294000+00:00\n",
      "@0084316d-91c3-4abc-9129-ecb53b55f9e2\tqs:f:16,4056\tch:i:53\tst:Z:2023-03-02T14:14:23.831000+00:00\tRG:Z:741dfaa4-41ba-469b-a04f-bc0c587b0122_dna_r10.4.1_e8.2_400bps_sup@v5.2.0_SQK-NBD114-24_barcode01\tPU:Z:AWJ804\tDT:Z:2023-03-02T14:01:04.294000+00:00\n",
      "@027eab1d-27d6-47eb-a9ff-24d2ae585b13\tqs:f:16,6151\tch:i:12\tst:Z:2023-03-02T14:19:06.755000+00:00\tRG:Z:741dfaa4-41ba-469b-a04f-bc0c587b0122_dna_r10.4.1_e8.2_400bps_sup@v5.2.0_SQK-NBD114-24_barcode01\tPU:Z:AWJ804\tDT:Z:2023-03-02T14:01:04.294000+00:00\n",
      "@0248c936-a0d8-4d95-bc96-01ddf34e47c6\tqs:f:19,9319\tch:i:108\tst:Z:2023-03-02T14:11:19.063000+00:00\tRG:Z:741dfaa4-41ba-469b-a04f-bc0c587b0122_dna_r10.4.1_e8.2_400bps_sup@v5.2.0_SQK-NBD114-24_barcode01\tPU:Z:AWJ804\tDT:Z:2023-03-02T14:01:04.294000+00:00\n"
     ]
    }
   ],
   "source": [
    "sed -n '1,40{1~4p}' ./BC1_notrim.fastq\n",
    "\n",
    "# this can also be combined to get a list of the first 10 annotation lines.\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c12a002c-f400-4a0b-a626-b02a1a1af587",
   "metadata": {},
   "source": [
    "### 3.2 SED ans fastq files\n",
    "How must the ```sed``` command from the cell above be modified to get a list of the first 15 nucleotide sequences of the *BC1_notrim.fastq* file?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d99ba55a-2bf0-4fa8-acdc-d452e7852ec0",
   "metadata": {},
   "outputs": [],
   "source": [
    "# your command"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "12092606-bdc2-413f-a39b-693a33af681a",
   "metadata": {},
   "source": [
    "### 3.3 Nanoplot\n",
    "Analyse the *BC1_notrim.fastq* file using ```NanoPlot``` \n",
    "- add the prefix *BC1notrim_* to all output files\n",
    "- generate ```kde plots```\n",
    "- change the color to blue\n",
    "- use log scale on the axis\n",
    "- add the title *BC1_sup_notrim* to the plots\n",
    "- write the output into a directory called *./nanoplot* which is on the same levels as *./data* in your directory structure\n",
    "- ```--fastq``` ./data/BC1_notrim.fastq\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2d610312-4c83-49eb-957a-34e470943468",
   "metadata": {},
   "source": [
    "<div class=\"alert alert-block alert-info\"> \n",
    "<b>NOTE</b> If you are unsure how to use Nanoplot to achieve these results, you can run Nanoplot with the `-h` flag to view a list of available options, or refer to the GitHub documentation.  \n",
    "</div>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "id": "e3cd25ff-c69f-4b77-be75-9832ecb2567b",
   "metadata": {},
   "outputs": [],
   "source": [
    "# go back one level in the directory structure\n",
    "cd .."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "id": "aadfdc71-6c7a-4740-9784-ff45e6e0df46",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WARNING:root:Plotly could not fetch or find Chrome automatically. Static exports may fail unless BROWSER_PATH is set. Details: [Errno 30] Read-only file system: '/opt/jupyter/lib/python3.11/site-packages/choreographer/cli/browser_exe'\n"
     ]
    }
   ],
   "source": [
    "# use this as a starting point\n",
    "NanoPlot -o ./nanoplot --fastq ./data/BC1_notrim.fastq"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f309f3b8-8e90-4db1-afe3-48605c24077e",
   "metadata": {},
   "source": [
    "### 3.4 Use less to show NanoStats\n",
    "change to the terminal and use ```less``` to open the *BC1notrim_NanoStats.txt*\n",
    "- What is the mean read length\n",
    "- What is the mean read Q-Score\n",
    "- What is the highest Q-score a Read achieved\n",
    "  "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c53676e9-2dc5-454c-a7f0-d9bc5ce288ad",
   "metadata": {},
   "source": [
    "# 4. Repeat QC for BC1_trimed.fastq\n",
    "Repeat all QC-Steps for *./data/BC1_trimed.fastq*\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
